clippy_lints/src/operators/arithmetic.rs

```diff
@@ -50,36 +52,82 @@ impl Arithmetic {
         span_lint(cx, ARITHMETIC, expr.span, "arithmetic detected");
         self.expr_span = Some(expr.span);
     }
+
+    /// Checks assign operators (+=, -=, *=, /=) already handled by the CTFE.
+    fn has_valid_assign_op(op: &Spanned<hir::BinOpKind>, rhs: &hir::Expr<'_>) -> bool {
+        match op.node {
```

Comment:
This could do with a check that the types involved are integral, as it could cause a panic for custom types. The comment should also be updated to say it's checking for cases that can't cause overflow rather than that are covered by CTFE